x-pack/plugins/ml/server/models/data_recognizer/data_recognizer.ts

```diff
@@ -418,6 +477,21 @@ export class DataRecognizer {
 
       datafeeds.push(...(await Promise.all(tempDatafeed)).filter(isDefined));
     }
+    // load the logoFile
+    if (module.logoFile !== undefined) {
+      try {
+        const logoFileString = await this._readFile(
+          `${this._modulesDir}/${dirName}/${module.logoFile}`
+        );
+        const logoFileIcon = JSON.parse(logoFileString);
+        module.logo = logoFileIcon?.icon;
```

Comment:
the `logo` property needs to be of type [`Logo`](https://github.com/elastic/kibana/blob/96739aad5c94e649beccf85ee219be73fb0efdd7/x-pack/plugins/ml/common/types/modules.ts#L80), as in the whole contents of the file, rather than just the icon.
The logo loading code in `findMatches` could be moved to a shared function to be used here.

Reply:
This should have been documented in the code.
A module can either contain a `logo` or `logoFile`. Originally the logo files contained the actual logo in svg. but later the apache and nginx logos were added to eui so we switched to just the IDs.
If `logo` does not exist in the module, it needs to be populated by the endpoint which is what `findMatches` does.

I think pretty much all of the modules on disc could now just contain a `logo` and no `logoFile`. Over the years, as  people have added more over modules they have copied existing ones and added a logo file rather than logo.

Because you need to check for 'logo`, your newly added `_loadLogoFile` function will need to also contain the check that logo exist, this can be moved from `findMatches`.
Also I think the code might be a bit cleaner if `_loadLogoFile` took the whole `module` as a parameter, maybe even the whole manifest too.

Reply:
As per our discussion - added the changes suggested above in 1022636da18fcf9a6562623b89f555171bfbf601 and will follow up with type updates and some clean up for modules specific code.